analyzer/src/main/kotlin/managers/Pip.kt

```diff
@@ -751,3 +751,9 @@ private fun Package.enrichWith(other: Package?): Package =
     } else {
         this
     }
+
+// Normalize all PyPI package names to be lowercase and hyphenated as per PEP 426:
+// "All comparisons of distribution names MUST be case insensitive,
+// and MUST consider hyphens and underscores to be equivalent".
+// https://www.python.org/dev/peps/pep-0426/#name  
+private fun String.normalizePackageName(): String = toLowerCase().replace('_', '-').replace('.', '-')
```

Comment:
The docs say nothing about considering also `.` and `-` to be equivalent, so where does that come from?

Nit: The comment should be turned into KDoc with a `/** ... */`-style multi-line comment.

Reply:
Update the comment to included PEP 503 which explains this

Reply:
Actually, PEP 503 suggests to use this code:

```
def normalize(name):
    return re.sub(r"[-_.]+", "-", name).lower()
```

Which means that also multiple consecutive `-` (or the other chars) will be collapsed to a single `-`, which is not something your current implementation does. Should we change to use a Regex, too?

Reply:
... and do the `toLowerCase()` as the last step, too, as the string might be shorter then 😁

Reply:
@sschuberth  If you can provide me with the Kotlin code to multiple consecutive `-` (or the other chars), I am happy to make the update.

Reply:
It's mostly just copy & paste of the Python code, and you can easily play around at the Kotlin playground, see

https://pl.kotl.in/bIoKb45gN

Reply:
Did not know about Kotlin playground thanks for showing me, fixed.